sample-macos-app/sample-macos-app/ImageLoading.swift

```diff
@@ -0,0 +1,36 @@
+import Cocoa
+
+public extension NSImageView {
+    private var activeTask: URLSessionDataTask? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKey.task) as? URLSessionDataTask
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKey.task, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
In general, it is better to use `OBJC_ASSOCIATION_RETAIN`. In this case, this will work properly when you set/get the value from different threads.

Reply:
This supposed to be called only from main thread. Otherwise it need a proper synchronization. Actually I took the idea from Alamofire image loader.